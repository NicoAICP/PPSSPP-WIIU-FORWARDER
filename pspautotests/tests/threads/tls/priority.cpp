#include "shared.h"

extern "C" int main(int argc, char *argv[]) {
	checkpointNext("Normal (FIFO):");
	{
		SceUID tls = sceKernelCreateTlspl("tls", PSP_MEMORY_PARTITION_USER, 0, 0x100, 1, NULL);
		void *data = sceKernelGetTlsAddr(tls);
		TlsplWaitThread wait1("waiting thread 1", tls, 300, 0x30);
		TlsplWaitThread wait2("waiting thread 2", tls, 300, 0x34);
		TlsplWaitThread wait3("waiting thread 3", tls, 300, 0x31);
		sceKernelFreeTlspl(tls);
		schedfTlspl(tls);
		sceKernelDelayThread(200000);
		sceKernelDeleteTlspl(tls);
	}

	checkpointNext("Priority:");
	{
		SceUID tls = sceKernelCreateTlspl("tls", PSP_MEMORY_PARTITION_USER, PSP_TLSPL_ATTR_PRIORITY, 0x100, 1, NULL);
		void *data = sceKernelGetTlsAddr(tls);
		TlsplWaitThread wait1("waiting thread 1", tls, 300, 0x30);
		TlsplWaitThread wait2("waiting thread 2", tls, 300, 0x34);
		TlsplWaitThread wait3("waiting thread 3", tls, 300, 0x31);
		sceKernelFreeTlspl(tls);
		schedfTlspl(tls);
		sceKernelDelayThread(200000);
		sceKernelDeleteTlspl(tls);
	}
	return 0;
}
