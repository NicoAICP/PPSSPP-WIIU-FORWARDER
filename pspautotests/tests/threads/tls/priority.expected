[x] Normal (FIFO):
[r]   ** started waiting thread 1
[r]   ** started waiting thread 2
[r]   ** started waiting thread 3
Tlspl: OK (size=60,name=tls,attr=00000000,index=0,blockSize=00000100,totalBlocks=00000001,freeBlocks=00000000,wait=2)
[r]   ** waiting thread 1 got result: 09d35700, received = ptr
[r]   ** waiting thread 1 delayed: 00000000
[r]   ** waiting thread 1 freed: 8002019d
[r]   ** waiting thread 2 got result: 09d35700, received = ptr
[r]   ** waiting thread 2 delayed: 00000000
[r]   ** waiting thread 2 freed: 8002019d
[r]   ** waiting thread 3 got result: 09d35700, received = ptr
[r]   ** waiting thread 3 delayed: 00000000
[r]   ** waiting thread 3 freed: 8002019d

[x] Priority:
[r]   ** started waiting thread 1
[r]   ** started waiting thread 2
[r]   ** started waiting thread 3
Tlspl: OK (size=60,name=tls,attr=00000100,index=0,blockSize=00000100,totalBlocks=00000001,freeBlocks=00000000,wait=2)
[r]   ** waiting thread 1 got result: 09d35700, received = ptr
[r]   ** waiting thread 1 delayed: 00000000
[r]   ** waiting thread 1 freed: 8002019d
[r]   ** waiting thread 3 got result: 09d35700, received = ptr
[r]   ** waiting thread 3 delayed: 00000000
[r]   ** waiting thread 3 freed: 8002019d
[r]   ** waiting thread 2 got result: 09d35700, received = ptr
[r]   ** waiting thread 2 delayed: 00000000
[r]   ** waiting thread 2 freed: 8002019d
